src/Builder/Blueprint.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace RstGroup\ObjectBuilder\Builder;
+
+use RstGroup\ObjectBuilder\Builder;
+
+final class Blueprint implements Builder
+{
+    /** @var Builder\Blueprint\Factory */
+    private $blueprintFactory;
+    /** @var ParameterNameStrategy */
+    private $strategy;
+
+    /** @codeCoverageIgnore */
+    public function __construct(
+        Builder\Blueprint\Factory $factory,
+        ParameterNameStrategy $strategy
+    ) {
+        $this->blueprintFactory = $factory;
+        $this->strategy = $strategy;
+    }
+
+    /** @param mixed[] $data */
+    public function build(string $class, array $data): object
+    {
+        $blueprint = $this->blueprintFactory->create($class);
+
+        $preparedData = $this->prepareData($data);
+
+        return $blueprint($preparedData);
+    }
+
+    /**
+     * @param mixed[] $data
+     * @return mixed[]
+     */
+    private function prepareData(array $data): array
+    {
+        $preparedData = [];
+
+        foreach ($data as $key => $value) {
+            if (is_array($value)) {
+                $value = $this->prepareData($value);
+            }
+
+            if (!is_int($key)) {
```

Comment:
Key can any, not only int or string

Reply:
The key can either be an integer or a string. The value can be of any type.
Other types are casted underlying.